src/main/java/com/google/sps/data/JourneyHandler.java

```diff
@@ -26,19 +27,29 @@
  * Class that stores and creates journeys for delivery slots.
  */
 public class JourneyHandler {
+  private PathFinder pathFinder;
+
+  public JourneyHandler(PathFinder pathFinder) {
+    this.pathFinder = pathFinder;
+  }
   /**
    * Finds a set of orders and an optimal journey for deliverySlot.
    */
-  public void processDeliveryRequest(DeliverySlot deliverySlot) throws ApiException, IOException, InterruptedException {
-    DeliverySystem deliverySystem = new DeliverySystem();
-    ArrayList<Point> orderedWaypoints = deliverySystem.findOptimalDeliveryJourneyForDeliverySlot(deliverySlot);
-    addDeliveryJourney(orderedWaypoints, deliverySlot);
+  public void processDeliveryRequest(DeliverySlot deliverySlot) throws ApiException, BadRequestException, DataNotFoundException, EntityNotFoundException, IOException, InterruptedException {
+    DeliverySystem deliverySystem = new DeliverySystem(pathFinder);
+    // Assign orders for the slot and create a journey with them. 
+    Journey journey = deliverySystem.createJourneyForDeliveryRequest(deliverySlot);
+    OrderHandler orderHandler = new OrderHandler(pathFinder);
+    orderHandler.updateStatusForOrders(journey.getOrders(), OrderHandler.OrderStatus.ASSIGNED.toString());
+    addDeliveryJourney(journey, deliverySlot);
   }
 
   /**
    * Adds the details of the delivery journey assigned to deliverySlot to datastore.
    */
-  protected void addDeliveryJourney(ArrayList<Point> waypoints, DeliverySlot deliverySlot) {
+  protected void addDeliveryJourney(Journey journey, DeliverySlot deliverySlot) {
```

Comment:
protected should rarely be used over private. Only when you want to allow inheritance to overwrite it.

Reply:
I changed `addDeliveryJourney` to private for now, but if I want to test the method, is it okay to make it public?

Reply:
In general, it's a bad idea to test implementation details, and private means something is an implementation detail. It's generally recommended to test all your public methods, since public methods are what the class advertises to external callers that they can and should call.

In your case processDeliveryRequest doesn't really do much other than call into other well tested methods. So I'd actually consider removing processDeliveryRequest altogether and just keeping a public `addDeliveryJourney`.

You might consider calling `orderHandler.updateStatusForOrders` inside `addDeliveryJourney`, as it doesn't really ever make sense to do these 2 actions separately (they should actually eventually happen within the same transaction, and having a transaction spawning multiple methods/classes makes things actually quite complicated).